n_helpers.c

```diff
@@ -150,60 +287,87 @@ const char *NoteBinaryEncode(const uint8_t *inBuf, uint32_t inLen,
+
+  @param  unencodedLength The length of an unencoded buffer.
+
   @returns  The max required buffer size to hold the encoded data.
-*/
+ */
 /**************************************************************************/
-uint32_t NoteBinaryEncodedMaxLength(uint32_t len)
+uint32_t NoteBinaryMaxEncodedLength(uint32_t unencodedLength)
 {
-    return cobsEncodedMaxLength(len);
+    return cobsEncodedMaxLength(unencodedLength);
 }
 
 //**************************************************************************/
 /*!
   @brief  Receive a large binary object from the Notecard's binary buffer
+
   @param  buffer A buffer to hold the binary object
   @param  bufLen The total length of the provided buffer
-  @param  dataLen An out parameter to hold the length of the decoded data in the
-                  output buffer.
+  @param  decodedOffset The offset to the decoded binary data already residing
+                        on the Notecard
+  @param  decodedLen [in/out] The length of the decoded data to fetch from the
+                     Notecard. If you wish to fetch the entire buffer from the
+                     given offset, set this value to `NOTE_C_BINARY_RX_ALL`.
+                     This parameter will return the bytes actually received from
+                     the Notecard.
+
   @returns  NULL on success, else an error string pointer.
-  @note  Buffers are decoded in place. The original contents of the buffer
-         will be modified.
-  @note  To determine the necessary size for the buffer, use
-         NoteBinaryRequiredRxBuffer.
-*/
+
+  @note  The buffer must be large enough to hold the encoded value of the
+         data store contents from the requested offset for the specified length.
+  @note  To determine the necessary buffer size for a given data length, use
+         (`NoteBinaryMaxEncodedLength()` + 1), or use
+         `NoteBinaryDataEncodedLength()` if you wish to consume the entire buffer.
+ */
 /**************************************************************************/
-const char * NoteBinaryReceive(uint8_t * buffer, size_t bufLen,
-                               size_t * dataLen)
+const char * NoteBinaryReceive(uint8_t * buffer, uint32_t bufLen,
+                               uint32_t decodedOffset, uint32_t * decodedLen)
 {
-    size_t requiredRxBufLen = 0;
-    const char *err = NoteBinaryRequiredRxBuffer(&requiredRxBufLen);
-    if (err) {
-        return err;
+    // Validate parameter(s)
+    if (!buffer) {
+        NOTE_C_LOG_ERROR("NULL buffer");
+        return ERRSTR("NULL buffer", c_err);
     }
-    if (!requiredRxBufLen) {
-        NOTE_C_LOG_ERROR("no data on notecard");
-        return ERRSTR("no data on notecard", c_err);
+    if (!decodedLen) {
+        NOTE_C_LOG_ERROR("decodedLen cannot be NULL");
+        return ERRSTR("decodedLen cannot be NULL", c_err);
     }
-    if (requiredRxBufLen > bufLen) {
+    if (bufLen < (cobsEncodedMaxLength(*decodedLen) + sizeof(char))) {
```

Comment:
In the doc above we say +1 but here we do + `sizeof(char)`. I think either is fine (slight preference for +1 -- it's easier to read and is there really a platform where `sizeof(char)` is not 1?), but I would be consistent.

Reply:
Just went with +1